tests/bal_business_logic_suite.py

```diff
@@ -24,44 +24,112 @@
 import os
 
 from openassetio import Context, TraitsData
-from openassetio.traits.managementPolicy import ManagedTrait
 from openassetio.test.manager.harness import FixtureAugmentedTestCase
 
+
 __all__ = []
 
 
-class Test_managementPolicy_default_behavior(FixtureAugmentedTestCase):
+class LibraryOverrideTestCase(FixtureAugmentedTestCase):
     """
-    Tests that by default, the BAL manages all entities, for read, but
-    nothing for write.
+    Utility base class to load a custom BAL library for the duration of
+    the tests in a subclass.
+
+    Assumes a `_library` class attribute that gives the name of the JSON
+    file to load from the `resources` directory.
     """
 
-    __trait_sets = (
-        {"blob", "image"},
-        {"container", "shot", "frame_ranged"},
-        {"definitely_unique"},
-    )
+    _library = None
+
+    def setUp(self):
+        # TODO(TC): Extend openassetio.test.manager to allow settings to
+        # be varied per test class/method.
+        # TODO(DF): Extend BAL to support supplying a library dict in
+        # settings, rather than just the path to a JSON file. Hence
+        # avoid re-reading from disk for every test.
```

Comment:
[question] What "to do"? The first TODO is moved from below and has lingered for some time. The second TODO I added as part of this work.

Perhaps one or both should be a [plan] and an issue created and these TODOs can be removed?

The second TODO could have been done as part of this work, but I thought it was a bit out of scope(?).

Reply:
Re. the first TODO, the reason to do it is if we expect that (groups of) tests frequently want to reinitialise the manager with different settings.  So we'd want a built-in supported way to do this via the test harness, rather than re-inventing this (somewhat hacky) wheel.

Until recently I would have said making issues for both of these is a no-brainer.  But I'm reluctant now.  @foundrytom, thoughts?

Reply:
The first one didn't have a justification in the wording, but it makes a huge difference to writing tests - the hoops you have to jump through to change the settings per test is onerous.

@feltech Im not sure about allowing a library to be passed through `settings`, as right now, that is an `InfoDict` which is deliberately flat, and doesn't support nesting.

Reply:
Added https://github.com/OpenAssetIO/OpenAssetIO/issues/775

Reply:
> Im not sure about allowing a library to be passed through `settings`, as right now, that is an `InfoDict` which is deliberately flat, and doesn't support nesting.

Ah, good point, I forgot that it was an InfoDict.  We could of course intercept the settings in Python before passing it along to C++, or serialise the library in some way (JSON, pickle).  But that does smell a little.  

Perhaps we need BAL SDK hooks, i.e. just accept that BAL is a puppet for testing and add some explicit mechanism that circumvents OpenAssetIO for doing testing fixture related things?

Anyway, for now I'll just remove both the TODOs.

Reply:
Removed the TODOs in 3f963b03